### DISCUSSION TITLE
Blink compat help

### DISCUSSION THREAD

**Discussion by DrummyFloyd** [LazyVim/LazyVim#4814] [Q&A] [answered]
hello , i'm migrating from `nvim-cmp` to `blink`
but i'm not sure how to handle some stuff
like adding ~~`cmp-emoji`~~ & `nvim-highlight-color` 

~~atm `cmp-emoji` seems to work, but i'm not sure if i override default configuration or `enabled_providers`~~
+ dunno how to make it work "nvim-highlight-colors` 

```lua
return {
  { "saghen/blink.compat" },
  {
    "saghen/blink.cmp",
    dependencies = {
      { "hrsh7th/cmp-emoji", "brenoprata10/nvim-highlight-colors" },
    },
    -- missing this in order to properly add `cmp-emoji` 
    opts_extend = { "sources.completion.enabled_providers" },
    opts = {
      sources = {
        completion = {
          enabled_providers = { "emoji" },
        },
        providers = {
          emoji = {
            name = "emoji",
            module = "blink.compat.source",
          },
          highlight_css = {
            name = "nvim-highlight-colors",
            module = "blink.compat.source",
            opts = function(_, opts)
              table.insert(opts.formatting, {
                format = require("nvim-highlight-colors").format,
              })
            end,
          },
        },
      },
    },
  },
}

```

**Comment by jyuan0** [comment, marked as answer]
Hi, I recently migrated and from my understanding of blink, this might not be working because `nvim-highlight-colors` isn't actually a `nvim-cmp` source. Rather than adding it as a source like `cmp-emoji`, you call the `setup` and use the `format` function.

If you're on the main branch of `blink.cmp`, I'd try adding `nvim-highlight-colors` as a regular plugin (not a dependency - I don't think it's needed), calling the `setup` and using the `format` function in a custom component in blink.

**Reply by DrummyFloyd** [reply]
you're right seems to work without any further config
thnak you !!!

**Reply by jyuan0** [reply]
Cheers, I'm glad you got it working!
